### ISSUE TITLE
Compilation error CS0122: 'ThisAssembly' is inaccessible due to its protection level.

### ISSUE THREAD

**Issue by L00pZBreak3r** [microsoft/CsWin32#507]
## Actual behavior

_src\Microsoft.Windows.CsWin32\Generator.cs_ (2456):
```
using Stream? templateStream = Assembly.GetExecutingAssembly().GetManifestResourceStream($"{ThisAssembly.RootNamespace}.templates.{name.Replace('/', '.')}.cs");
```
**Error CS0122**: 'ThisAssembly' is inaccessible due to its protection level.

## Expected behavior

Successful compilation.

**Comment by AArnott** [comment]
The build succeeds, as [attested to by our CI](https://dev.azure.com/azure-public/winsdk/_build/latest?definitionId=47&branchName=main) as well as my own machine.

The error you're seeing might come up when you haven't restored packages first. I'm not sure what else may cause it. I'm going to close the issue because this isn't a legit break in the code, but I'm happy to keep conversing for a while to help you get unblocked.

**Comment by gerhard17** [comment]
Same problem here. 
Where should the ThisAssembly type be defined? 
VisualStudio finds an internal static class ThisAssembly in MessagePack assembly.

**Comment by gerhard17** [comment]
Found the problem:

ThisAssembly is a nuget package. 
This package is not directly included in the project file. 
I'm not sure, which other nuget package references the ThisAssembly package.
I played only with partial code, so this root package may be lost in my special case.

Solutions:

- Use the complete original package includes or 
- Reference ThisAssembly yourself (attention, the newest nuget seems to implement ThisAssembly in a slightly different way) or
- Replace the ThisAssembly.Rootnamespace with a fixed string.

Regards

**Comment by L00pZBreak3r** [comment]
_Replace the ThisAssembly.Rootnamespace with a fixed string._
It's already done with "Microsoft.Windows.CsWin32" 2 months ago.

**Comment by gerhard17** [comment]
Thanks you for your response!

But the change is at least not in the main trunk.

That said, I do not have any problems with this issue. I was only confused by the compiler error at first sight.

**Comment by AArnott** [comment]
> ThisAssembly is a nuget package

That is _not_ where the `ThisAssembly` type comes from. `ThisAssembly` comes from Nerdbank.GitVersioning, in our case, which generates the source for it during the build. If you're seeing a compile error in this repo regarding it, you've got some issue with your machine that I can't predict. If you send me a binlog (build from the command line with the /bl switch) I could perhaps figure out what's going wrong.

**Comment by L00pZBreak3r** [comment]
The only thing I'd like to know is if _ThisAssembly_ string is equal to **"Microsoft.Windows.CsWin32"**.

**Comment by AArnott** [comment]
`ThisAssembly` is not a string. It is a class.

**Comment by L00pZBreak3r** [comment]
Of course, I meant _ThisAssembly.RootNamespace_. Don't be so picky.

**Comment by AArnott** [comment]
I'm not trying to be picky. There are *many* strings fields on that class. You could have been referring to `AssemblyName` or many other things. I haven't been studying this long discussion because I gave my answer far above, but I caught a mention of another nuget package that generates a ThisAssembly class, so when you were asking about a string, I wondered if this was some odd thing from that other nuget package. I actually take quite a bit of time to help customers get things unblocked across github, and I'm sorry this came across as picky. It wasn't meant that way. 

As for what the value of RootNamespace is, yes I suspect that is its value. But you could always verify that by looking at the msbuild properties or deriving its default value from the project name.

**Comment by L00pZBreak3r** [comment]
I also would like to apologize if I happened to sound ungrateful. Yes, of course _ThisAssembly_ is a class, and I should have been more careful when I wrote that question.

I solved this ThisAssembly issue with replacing **ThisAssembly.RootNamespace** with string **"Microsoft.Windows.CsWin32"**. It’s the only place where ThisAssembly is referenced. So, using a constant string here allows to get rid of using the ThisAssembly completely. It’s not referenced anywhere else.
